### PR TITLE
feat(tier4_traffic_light_rviz_plugin): add traffic light publish panel

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Related links

- https://github.com/autowarefoundation/autoware.universe/issues/533

## Tests performed

- please check traffic light signals are published correctly.
- topic name: `/perception/traffic_light_recognition/traffic_signals`
- See the following video for instruction on how to use this.

https://user-images.githubusercontent.com/44889564/161898028-04690740-d9a4-4cb2-b31c-9e592fc04203.mp4

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by yukkysaito** [comment]
@satoshi-ota Good!

I have one question :pray: 
Does it publish as long as the publish button is held down?
If there is a reset button, I think it is better to continue to publish button when clicked.

**Comment by yukkysaito** [comment]
cc @aohsato @miursh

**Comment by satoshi-ota** [comment]
@yukkysaito 

> Does it publish as long as the publish button is held down?

Yes. it does.

> If there is a reset button, I think it is better to continue to publish button when clicked.

It has already been designed that plugin continues to publish signals while `PUBLISH` button is pushed, and the button is checkable (Keep clicked state until click once more).

**Comment by yukkysaito** [comment]
@satoshi-ota In the video,  it seems to not publish when the button is released. 
is it different now?

**Comment by satoshi-ota** [comment]
> @satoshi-ota In the video, it seems to not publish when the button is released. is it different now?

Oh, I misunderstood your comment :pray: As you said, it stop publishing when `PUBLISH` button is released. On the other hand, I designed `RESET` button not to stop publishing but to clear traffic light signal array.

**Comment by satoshi-ota** [comment]
@yukkysaito please check behavior of the panel :pray: 

https://user-images.githubusercontent.com/44889564/162093306-9eaba5e7-74f0-4a61-932b-7b79f76632bd.mp4

**Comment by yukkysaito** [comment]
LGTM :+1:

**Review by yukkysaito — DISMISSED** [review]

File: common/tier4_traffic_light_rviz_plugin/src/traffic_light_publish_panel.cpp [hunk 1]
```diff
@@ -0,0 +1,281 @@
+//
+//  Copyright 2020 Tier IV, Inc. All rights reserved.
```
Comment (yukkysaito) [inline common/tier4_traffic_light_rviz_plugin/src/traffic_light_publish_panel.cpp:2]:
```suggestion
//  Copyright 2022 TIER IV, Inc. All rights reserved.
```

File: common/tier4_traffic_light_rviz_plugin/src/traffic_light_publish_panel.hpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
+//
+//  Copyright 2020 Tier IV, Inc. All rights reserved.
```
Comment (yukkysaito) [inline common/tier4_traffic_light_rviz_plugin/src/traffic_light_publish_panel.hpp:2]:
```suggestion
//  Copyright 2022 TIER IV, Inc. All rights reserved.
```

**Review by yukkysaito — APPROVED** [review]

File: common/tier4_traffic_light_rviz_plugin/src/traffic_light_publish_panel.cpp [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+    traffic_light.color = TrafficLight::DOWN_ARROW;
+  } else if (color == "DOWN_LEFT_ARROW") {
+    traffic_light.color = TrafficLight::DOWN_LEFT_ARROW;
+  } else if (color == "DOWN_RIGHT_ARROW") {
+    traffic_light.color = TrafficLight::DOWN_RIGHT_ARROW;
+  } else if (color == "FLASHING") {
+    traffic_light.color = TrafficLight::FLASHING;
+  } else if (color == "UNKNOWN") {
+    traffic_light.color = TrafficLight::UNKNOWN;
+  }
```
Comment (isamu-takagi) [inline common/tier4_traffic_light_rviz_plugin/src/traffic_light_publish_panel.cpp:133]: I think the correspondence is:
```
color = {UNKNOWN, RED, AMBER, AMBER, WHITE}
shape = {UNKNOWN, CIRCLE, LEFT_ARROW, RIGHT_ARROW, UP_ARROW, DOWN_ARROW, DOWN_LEFT_ARROW, DOWN_RIGHT_ARROW, CROSS}
status = {UNKNOWN, SOLID_OFF, SOLID_ON, FLASHING}
```

**Pull request merged**